dirty_cat/datasets/tests/test_fetching.py

```diff
@@ -32,6 +32,23 @@
 from dirty_cat.datasets._utils import get_data_dir as _get_data_dir
 
 
+class TemporaryDirectory(_TemporaryDirectory):
```

Comment:
Do we really need this? Can we not add the "Path" later in the enclose code? Or use the required functionality outside of Path?

Reply:
It looks to me that if we don't want to subclass like it's currently done, we either have to add support for `str` paths in the fetching code, or add boilerplate where it's called in the tests like that:

From

```py
with TemporaryDirectory() as temp_dir:
    ...
```

to

```py
with TemporaryDirectory() as temp_dir:
    temp_dir = Path(temp_dir).resolve()
    ...
```

which IMO is cumbersome and a bit error-prone.

Note: I though of using a function such as
```py
def get_temp_dir() -> Path:
    with TemporaryDirectory() as temp_dir:
        return Path(temp_dir).resolve()
```

but first, we lose the `with` ability with the function (can't do `with get_temp_dir() as temp_dir`), and most importantly, as soon as the path is returned, the `with` in the func exists, thus the directory is removed and can't be used afterwards.

Reply:
Maybe the best solution would be to refactor the fetching code, as a parallel issue solved by #432 would allow users to pass a directory at fetching time, and while using pathlib is good practice, I don't think it's common practice for our userbase. Most people probably work with `os.path` and thus use strings.